simpleclient/src/main/java/io/prometheus/client/CollectorRegistry.java

```diff
@@ -125,31 +125,53 @@ public void clear() {
   public Enumeration<Collector.MetricFamilySamples> metricFamilySamples() {
     return new MetricFamilySamplesEnumeration();
   }
+
+  public Enumeration<Collector.MetricFamilySamples> metricFamilySamples(Set<String> includedNames) {
```

Comment:
I'm not sure what you mean. Are you referring to Prometheus server development?

Are you saying the we might need regex matching for the `names[]` parameters instead of just the simple `contains` check I went with, or should I hold off on that? For now I was just mimicking the Python implementation (though I confess I'm not very familiar with Python)

Do you want me to rename the method to `filteredMetricFamilySamples`?